src/App.tsx

```diff
@@ -1,223 +1,228 @@
   const newTodoField = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => (setErrorMessage('Unable to load a todos')));
+    }
+
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  const onClickAll = () => {
+    setFilterTodos('All');
+  };
+
+  const onClickActive = () => {
+    setFilterTodos('Active');
+  };
+
+  const onClickCompleted = () => {
+    setFilterTodos('Completed');
+  };
+
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle) {
+      setErrorMessage('Title can not be empty');
+
+      return;
+    }
+
+    if (user) {
+      setTempTodo({
+        id: 0,
+        title: todoTitle,
+        completed: false,
+        userId: user.id,
+      });
+
+      setIsLoading(true);
+      addTodo(todoTitle, user.id)
+        .then(response => {
+          setTodos(prev => [...prev, {
+            id: response.id,
+            title: response.title,
+            completed: response.completed,
+            userId: response.userId,
+          }]);
+        })
+        .catch(() => setErrorMessage('Unable to add a todo'))
+        .finally(() => {
+          setIsLoading(false);
+          setTempTodo(null);
+          setTodoTitle('');
+        });
+    }
+  };
+
+  const onDeleteTodo = (id: number) => {
+    setIsLoading(true);
+    deleteTodo(id)
+      .then(() => (
+        setTodos(currentTodos => currentTodos
+          .filter(todo => todo.id !== id))
+      ))
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+      });
+    setIsLoading(false);
+  };
+
+  const onUpdateTodo = (chosenTodo: Todo) => {
+    setTodosToUpdate(prev => [...prev, chosenTodo]);
+
+    updateTodo(chosenTodo.id, chosenTodo.title, chosenTodo.completed)
+      .then((udpatedTodo) => {
+        setTodos(currentTodos => currentTodos.map(todo => (
+          todo.id !== udpatedTodo.id
+            ? todo
+            : {
+              id: udpatedTodo.id,
+              userId: udpatedTodo.userId,
+              title: udpatedTodo.title,
+              completed: udpatedTodo.completed,
+            }
+        )));
+      })
+      .catch(() => {
+        setErrorMessage('Unable to update a todo');
+      })
+      .finally(() => {
+        setTodosToUpdate([]);
+      });
+  };
+
+  const onToggleAll = () => {
+    setIsLoading(true);
+    if (isToggledAll === false) {
+      todos.forEach(todo => {
+        if (!todo.completed) {
+          onUpdateTodo({ ...todo, completed: !todo.completed });
+        }
+      });
+      setIsToggledAll(true);
+    } else {
+      todos.forEach(todo => {
+        onUpdateTodo({ ...todo, completed: !todo.completed });
+      });
+      setIsToggledAll(false);
+    }
+
+    setIsLoading(false);
+  };
+
+  const onClickClearCompleted = () => {
```

Comment:
All callbacks that we are passing to another component  should be wrapped with `useCallback`